cpp/src/json/json_path.cu

```diff
@@ -600,7 +600,8 @@ class path_state : private parser {
  private:
   cudf::io::parse_options_view json_opts{',', '\n', '\"', '.'};
 
-  bool parse_path_name(string_view& name, string_view const& terminators)
+  // inside_brackets is set to true while parsing values enclosed within [ ] example: ['book']
+  bool parse_path_name(string_view& name, string_view const& terminators, bool inside_brackets)
```

Comment:
We often use `enum class` instead of `bool` arguments in libcudf. Example: https://github.com/rapidsai/cudf/blob/63e9040d0e80a8ccdb52892bfe10a99309d8b2d5/cpp/include/cudf/types.hpp#L121-L127

I think that would be a sensible thing to do here, and the clarity of the naming would eliminate the need for the awkward inline comments describing the boolean values like `/*inside brackets*/` above.

Something like:
```cpp
/** 
 * @brief Enum to specify whether parsing values enclosed within brackets, like `['book']`
 */
enum class bracket_state : bool {
  INSIDE,   ///< Parsing inside brackets
  OUTSIDE   ///< Parsing outside brackets
}; 
```

Reply:
Thank you for the thorough instructions, I have updated the PR!